src/oneNoteSingleNotebookDropdown.tsx

```diff
@@ -32,6 +37,26 @@ export class OneNoteSingleNotebookDropdown extends React.Component<OneNoteSingle
 		});
 	}
 
+	private setWrapperRef(node) {
+		this.wrapperRef = node as Node;
+	}
+
+	private handleClickOutside(event) {
+		if (this.wrapperRef && !this.wrapperRef.contains(event.target) && this.state.popupVisible) {
```

Comment:
Instead of checking if the parent contains the click, why not check if you (the dropdown) don't contain the click? That would prevent us from having to pass in a ref.

Reply:
We aren't passing a ref down from the parent. 'Ref' actually refers to a React concept (a method that executes on mounting and unmounting). Here we attach the ref to the wrapping div element that this component renders (so it has no knowledge of its parent)